### PR TITLE
Make it possible to turn off HTTP/2 support

### PR DESCRIPTION
**Pull request by swankjesse**
We're running into problems with HTTP/2 and rebalancing

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt [hunk 1]
```diff
@@ -96,16 +109,34 @@ class JettyService @Inject internal constructor(
...
+        }
+        httpsConnectionFactories += http2
       }
+
       val http1 = HttpConnectionFactory(httpsConfig)
-      val httpsConnector = ServerConnector(server, null, null, null,
-          webConfig.acceptors ?: -1, webConfig.selectors ?: -1, ssl, alpn, http2, http1)
+      httpsConnectionFactories += http1
+
+      val httpsConnector = ServerConnector(
```
Comment (zhxnlai) [inline misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: This `startUp` method is long. I would move this line to a `createHttpsConnector` method (same for `createHttpConnector`)

File: misk/src/main/kotlin/misk/web/WebConfig.kt [hunk 2]
```diff
@@ -6,21 +6,44 @@ import misk.security.ssl.TrustStoreConfig
...
+
+  /** If a connection is unused for this many milliseconds, it is closed. */
   val idle_timeout: Long,
+
+  /** The network interface to bind to. Null or 0.0.0.0 to bind to all interfaces. */
   val host: String? = null,
+
   val ssl: WebSslConfig? = null,
+
+  /** HTTP/2 support is currently opt-in because we can't load balance it dynamically. */
```
Comment (alecholmes) [inline misk/src/main/kotlin/misk/web/WebConfig.kt]: `true` being the default makes HTTP/2 opt-out.

Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/WebConfig.kt]: Yep. I’m gonna flip in follow-up

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt (hunk 1 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: Lemme consider for follow-up. I’m not bothered by long plumbing methods.

**Pull request merged**